### ISSUE TITLE
Aarch32 builds fail with SIGILL on ARMv7

### ISSUE THREAD

**Issue by sappling** [AdoptOpenJDK/openjdk-build#970]
**Platform:**
Beagle Bone Black (ARMv7-A)
**Architecture:**
aarch32

Running almost anything other than just "java -version" seems to throw SIGILL.  This was tested with the March 17th build from [here](https://ci.adoptopenjdk.net/job/build-scripts/job/jobs/job/jdk11u/job/jdk11u-linux-arm-hotspot/).

Here are the results from a java -version and javac (with no args).
```
debian@beaglebone:~$ jdk-11.0.2+9/bin/java -version
openjdk version "11.0.3" 2019-04-16
OpenJDK Runtime Environment AdoptOpenJDK (build 11.0.3+9-201903171916)
OpenJDK Server VM AdoptOpenJDK (build 11.0.3+9-201903171916, mixed mode)

debian@beaglebone:~$ jdk-11.0.2+9/bin/javac
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGILL (0x4) at pc=0xb4204e88, pid=2082, tid=2083
#
# JRE version: OpenJDK Runtime Environment (11.0.3+9) (build 11.0.3+9-201903171916)
# Java VM: OpenJDK Server VM (11.0.3+9-201903171916, mixed mode, serial gc, linux-)
# Problematic frame:
# J 25 c2 java.util.HashMap.resize()[Ljava/util/HashMap$Node; java.base@11.0.3 (356 bytes) @ 0xb4204e88 [0xb4204da0+0x000000e8]
#
# No core dump will be written. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
#
# An error report file with more information is saved as:
# /home/debian/hs_err_pid2082.log
Could not load hsdis-arm.so; library not loadable; PrintAssembly is disabled
#
# If you would like to submit a bug report, please visit:
#   https://github.com/eclipse/openj9/issues
#
```

**Comment by sappling** [comment]
Note that we have a working aarch32 build we created ourselves directly from OpenJDK (without using AdoptOpenJDK scripts).  This was cross compiled from Ubuntu 10.04/x64.  I will get the engineer who worked on this to post more details about the working build.

**Comment by ClintonFlowers** [comment]
I'll be investigating more, but for comparison, the AdoptOpenJDK [is currently configuring with](https://ci.adoptopenjdk.net/job/build-scripts/job/jobs/job/jdk11u/job/jdk11u-linux-arm-hotspot/79/consoleText):

> --verbose --with-version-opt=201903201154 --without-version-pre --with-version-build=9 --with-vendor-version-string=AdoptOpenJDK --with-vendor-url=https://adoptopenjdk.net/ --with-vendor-name=AdoptOpenJDK --with-vendor-bug-url=https://github.com/AdoptOpenJDK/openjdk-build/issues --with-vendor-vm-bug-url=https://github.com/eclipse/openj9/issues --with-boot-jdk=/home/jenkins/bootjdks/jdk-10.0.2+13 --enable-ccache --with-jvm-variants=server --with-alsa=/home/jenkins/workspace/build-scripts/jobs/jdk11u/jdk11u-linux-arm-hotspot/workspace/./build//installedalsa --with-x=/usr/include/X11 --with-debug-level=release --with-native-debug-symbols=none --with-jobs=4 --with-memory-size=2000 --disable-warnings-as-errors --with-freetype=bundled --with-jobs=28

and

> Building targets 'images test-image' in configuration 'linux-arm-normal-server-release'

The VM which I have working was configured with:

> --openjdk-target=arm-linux-gnueabihf --with-abi-profile=arm-vfp-hflt --disable-warnings-as-errors --enable-unlimited-crypto --enable-headless-only=yes --with-native-debug-symbols=none --with-debug-level=release --with-jvm-variants=client

and

> Building target 'legacy-jre-image' in configuration 'linux-arm-normal-client-release'

or

> Building target 'jdk-image' in configuration 'linux-arm-normal-client-release'

I believe the most significant difference between the one we have working and the others is that I was using the Client JVM, not Server. Our reference hardware for the builds is the BeagleBone Black and/or Blue, and we are cross-compiling from a Ubuntu 18.04 (x64) LTS VM. 

One of our testers ran into a SIGILL error in late September while testing out our setup/configure/build script for a JDK 11 Server VM on his BeagleBone Black. However, at that time his 11 Server build was actually getting further than mine so I couldn't reproduce it.

**Comment by jgneff** [comment]
Try running JDK 11 with the option `-XX:-AssumeMP` as shown below.

**Fails:**
```ShellSession
$ ~/opt/jdk-11.0.3+7/bin/javac 
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGILL (0x4) at pc=0x2b875a40, pid=7837, tid=7838
...
```

**Works:**
```ShellSession
$ ~/opt/jdk-11.0.3+7/bin/javac -J-XX:-AssumeMP
OpenJDK Server VM warning: Option AssumeMP was deprecated in version 10.0 and
will likely be removed in a future release.
Usage: javac <options> <source files>
where possible options include:
...
```

See the issue I just opened as #1070 for details.

**Comment by sappling** [comment]
Thanks - that does get me running with JDK11.  Your new issue helped me understand more of what is going on here.

**Comment by aahlenst** [comment]
Closing this issue due to lack of activity. Furthermore, we made some improvements in the ARM area since the last post. Happy to reopen if this is still a problem.
